script/src/types.rs

```diff
@@ -1008,6 +1008,13 @@ where
     }
 }
 
+/// When the vm is initialized, arguments are loaded onto the stack.
+/// This enum specifies how to locate these arguments.
+pub enum VmArgs {
+    Reader(u64, u64, u64),
```

Comment:
Now that we are building a proper type here, please give those 3 fields distinctive names.